### PR TITLE
Moved decoders to separate module so they can be used externally

### PR DESCRIPTION
**Pull request by Kheldar**

### CODE REVIEW

File: src/Arbor/File/Format/Asif/Format/Decoder.hs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import qualified Data.Binary.Get                   as G
+import qualified HaskellWorks.Data.Network.Ip.Ipv4 as IP4
+import qualified HaskellWorks.Data.Network.Ip.Ipv6 as IP6
+
+getWord32x4 :: G.Get (Word32, Word32, Word32, Word32)
+getWord32x4 = do
+  a <- G.getWord32be
+  b <- G.getWord32be
+  c <- G.getWord32be
+  d <- G.getWord32be
```
Comment (AlexeyRaga) [inline src/Arbor/File/Format/Asif/Format/Decoder.hs]: Other cases use `getWord32le`, are you use `getWord32be` here deliberately?

Comment (Kheldar) [reply src/Arbor/File/Format/Asif/Format/Decoder.hs]: It's deliberate in the sense that Dave said they did it deliberately.

Comment (newhoggy) [reply src/Arbor/File/Format/Asif/Format/Decoder.hs]: Probably because `IPv6` is stored in Network Order, which is the same as BE.

Comment (newhoggy) [reply src/Arbor/File/Format/Asif/Format/Decoder.hs]: It's interesting though that we store IPv4 in LE, so there is some inconsistency there.

**Pull request merged**